github/ProjectCard.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2018 bbi-yggy <yossarian@blackbirdinteractive.com>                 #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+import json
+
+import github.GithubObject
+
+# NOTE: There is currently no way to get cards "in triage" for a project.
+# https://platform.github.community/t/moving-github-project-cards-that-are-in-triage/3784
+#
+# See also https://developer.github.com/v4/reference/object/projectcard for the next generation GitHub API,
+# which may point the way to where the API is likely headed and what might come back to v3. E.g. ProjectCard.content member.
+class ProjectCard(github.GithubObject.NonCompletableGithubObject):
```

Comment:
Shouldn't this be `CompletableGithubObject`?

Reply:
My understanding was that any class whose attributes could be fully retrieved from the results of the original request should be NonCompletable, since no additional API methods are needed. Which may actually suggest that Project should also be NonCompletable. From the information you cited earlier, "if for some reason an attribute accessed by the user has never been retrieved from the API, this class does an implicit request to retrieve all missing fields" - but for ProjectCard and ProjectColumn all the attributes are available after the first request.

I'm happy to be corrected if my understanding is incorrect.

Reply:
My understanding is that as long as we can lazy retrieve a resource (the response has the `url` attribute, kinda like a permanent link), then the corresponding GithubObject type should be `CompletableGithubObject`. You can make comparisons with Issue/PullRequests, they have a similar relationship with org/repos. Also an observation is that, most of the types in this library are `CompletableGithubObject`. Again, I did not author the original code so I may be wrong.

Reply:
Hmm, ok, I'll revisit this. Where in the API does "completability" get used?